### PR TITLE
continue spinlock

### PR DESCRIPTION
**Pull request by chenyukang**
hi, @ckkashyap ,

blocked by readeflags in x86asm.rs,  Do you know how to write this inline assembly code.

### CODE REVIEW

File: kernel/x86asm.rs [hunk 1]
```diff
@@ -21,19 +21,40 @@
...
-        asm!("inb %dx, %al" : "={ax}"(ret) : "{dx}"(port));
-	}	
-        return ret;
+        asm!("inb %dx, %al": "={ax}"(ret) : "{dx}"(port));
+    }
+    return ret;
+}
+
+
+pub fn readeflags() -> u32 {
```
Comment (chenyukang) [inline kernel/x86asm.rs]: need more digg

**Comment by chenyukang** [comment]
pushfl error: instruction requires: Not 64-bit mode

**Comment by ckkashyap** [comment]
I've added read_eflags function now

**Comment by ckkashyap** [comment]
the inline assembly in rust seems similar to gcc - http://doc.rust-lang.org/0.11.0/guide-unsafe.html#inline-assembly

**Pull request merged**